### PR TITLE
[ADD] Added option to stub api calls

### PR DESCRIPTION
**Pull request by andreyleonardo**
This is related to ticket #12

- Added option to stub api calls like the official gem.
- Some code style changes.

### CODE REVIEW

**Comment by whatthewhat** [comment]
@andreyleonardo this looks great! 👍 

I'll do a proper review later this week, but just a couple quick notes:

1. Can we make the logger configurable, but default to `Logger.new STDOUT`? Something like the example here https://github.com/whatthewhat/simple_segment/issues/4 (this is just a nice to have, I can add it later, though).
2. Also not a deal breaker, but if it's possible to squash the commits to 2: one that adds the feature and one that addresses the rubocop issues that would be awesome! (bonus points if the lines stay under 80 chars)

File: README.md [hunk 1]
```diff
@@ -74,6 +74,19 @@ analytics.batch do |batch|
...
+### Stub API calls
+
+You can stub your API calls avoiding unecessary calls in development and automated test environments.
+Backwards compatibility with the official gem
+
+```ruby
+analytics = SimpleSegment::Client.new({
+  write_key: 'YOUR_WRITE_KEY', # required
+  stub: true
+  }
```
Comment (whatthewhat) [inline README.md]: Looks like an extra `}` snuck in

**Comment by andreyleonardo** [comment]
1 - Sure. I'll change this. =)
2 - Sorry about the Rubocop. I use on my projects with 120 chars, I'll fix this and open a separated PR for those changes as you proposed.

**Comment by andreyleonardo** [comment]
Closing this PR to open 2 separated PRs.

**Pull request closed without merging**